airgun/entities/host_new.py

```diff
@@ -272,15 +435,146 @@ def get_parameters(self, entity_name):
         return view.parameters.read()
 
     def add_new_parameter(self, entity_name, parameter_name, parameter_type, parameter_value):
+        """
+        Function that adds new parameter to the host
+
+        Args:
+            entity_name: host on which we want to add parameter
+            parameter_name: Name of the parameter to be added
+            parameter_type: Type of the parameter to be added
+                [string, boolean, integer, real, array, hash, yaml, json]
+            parameter_value: Value of the parameter to be added
+
+        Raises:
+            ValueError: If parameter type is not valid
+            ValueError: If parameter with the same name already exists
+        """
+
+        # Sanitize input
+        parameter_type = parameter_type.lower()
+        available_types = ['string', 'boolean', 'integer', 'real', 'array', 'hash', 'yaml', 'json']
+        if parameter_type not in available_types:
+            raise ValueError(f'Parameter type {parameter_type} is not valid')
+
         view = self.navigate_to(self, 'NewDetails', entity_name=entity_name)
         view.wait_displayed()
         self.browser.plugin.ensure_page_safe()
+        view.parameters.click()
+        view.parameters.searchbar.fill(parameter_name)
+        view.wait_displayed()
+        # Check if parameter with given name does not exist. If it does, raise ValueError.
+        if view.parameters.parameters_table.row_count != 0:
+            if view.parameters.parameters_table[0][0].text == parameter_name:
+                raise ValueError(f'Parameter with name {parameter_name} already exists')
```

Comment:
from looking at the param search, this could yield multiple resutls if the searched param_name is a substring of other names. This could lead to false negative since you just check the first result here

Reply:
![image](https://user-images.githubusercontent.com/8287578/234562709-50658aa8-b86c-4f99-b8fd-825071bedab5.png)

Reply:
maybe you could request a normal key=value search at this tab, as it is on the insights tab, that would probably solve the issue as well, up to you

Reply:
key=value that is present on the insights tab is not working here sadly

Reply:
hmm. I was thinking about this problem with the substring. But when I was testing that if I search for the parameter (and it existed) satellite puts that result as first every time.

Reply:
update: Satellite puts it first when you have table ordering set to ascending

Reply:
Then you may have to cycle through all the results, checking for equality.

Reply:
What if
1) The searched term is something's suffix? I can imagine that works too since the search is only prefix-based?
2) You search for `text`, you find `textsuffix` and delete it in the deletion method? You really should check for equality.

Reply:
![image](https://user-images.githubusercontent.com/62888716/234578956-dbc68396-044d-4299-ab9d-9d0257655c40.png)